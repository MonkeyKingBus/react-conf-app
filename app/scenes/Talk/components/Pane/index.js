import React, { Component, PropTypes } from 'react';
import {
	Animated,
	PixelRatio,
	ScrollView,
	StyleSheet,
	Text,
	TouchableOpacity,
	View,
} from 'react-native';
import moment from 'moment';

import { TIME_FORMAT } from '../../../../constants';
import theme from '../../../../theme';
import Avatar from '../../../../components/Avatar';

import Preview from '../Preview';

export default class TalkPane extends Component {
	render () {
		const {
			nextTalk,
			nextTalkPreviewIsEngaged,
			onHeroLayout,
			prevTalk,
			prevTalkPreviewIsEngaged,
			showSpeakerModal,
			visibleTalk,
			...props
		} = this.props;

		const touchableProps = {
			activeOpacity: 0.66,
			onPress: showSpeakerModal,
		};

		return (
			<ScrollView style={{ flex: 1 }} scrollEventThrottle={90} ref="scrollview" {...props}>
				{!!prevTalk && (
					<View ref="prevTalkPreview" style={{ opacity: 0 }}>
						<Preview
							isActive={prevTalkPreviewIsEngaged}
							position="top"
							subtitle={`${moment(prevTalk.time.start).format(TIME_FORMAT)} - ${prevTalk.speaker.name}`}
							title={prevTalk.title}
						/>
					</View>
				)}
				<View style={styles.hero} onLayout={onHeroLayout}>
					<TouchableOpacity {...touchableProps}>
						<Animated.View style={styles.heroSpeaker}>
							<Avatar source={visibleTalk.speaker.avatar} />
							<Text style={styles.heroSpeakerName}>
								{visibleTalk.speaker.name}
							</Text>
							<Text style={styles.heroSpeakerHint}>
								(tap for more)
							</Text>
						</Animated.View>
					</TouchableOpacity>
					<Text style={styles.heroTitle}>
						{visibleTalk.title}
					</Text>
				</View>

				<View style={styles.summary} ref="summary">
					<Text style={styles.summaryText}>
						{visibleTalk.summary}
					</Text>
				</View>

				{!!nextTalk && (
					<View ref="nextTalkPreview" style={{ opacity: 0 }}>
						<Preview
							isActive={nextTalkPreviewIsEngaged}
							position="bottom"
							subtitle={`${moment(nextTalk.time.start).format(TIME_FORMAT)} - ${nextTalk.speaker.name}`}
							title={nextTalk.title}
						/>
					</View>
				)}
			</ScrollView>
		);
	}
};
TalkPane.propTypes = {
	nextTalk: PropTypes.object,
	nextTalkPreviewIsEngaged: PropTypes.bool,
	onHeroLayout: PropTypes.func, // isRequired
	onScroll: PropTypes.func, // isRequired
	onScrollEndDrag: PropTypes.func, // isRequired
	prevTalk: PropTypes.object,
	prevTalkPreviewIsEngaged: PropTypes.bool,
	showSpeakerModal: PropTypes.func, // isRequired
	visibleTalk: PropTypes.object, // isRequired
};

const styles = StyleSheet.create({
	hero: {
		alignItems: 'center',
		backgroundColor: 'white',
		borderBottomColor: theme.color.gray20,
		borderBottomWidth: 1 / PixelRatio.get(),
		borderTopColor: theme.color.gray20,
		borderTopWidth: 1 / PixelRatio.get(),
		marginTop: -(1 / PixelRatio.get()),
		paddingHorizontal: theme.fontSize.large,
		paddingBottom: theme.fontSize.xlarge,
	},
	heroSpeaker: {
		alignItems: 'center',
		paddingHorizontal: theme.fontSize.xlarge,
		paddingTop: theme.fontSize.xlarge,
	},
	heroSpeakerHint: {
		color: theme.color.gray40,
		fontSize: theme.fontSize.xsmall,
		paddingBottom: theme.fontSize.large,
	},
	heroSpeakerName: {
		color: theme.color.blue,
		fontSize: theme.fontSize.default,
		fontWeight: '500',
		marginTop: theme.fontSize.small,
	},
	heroTitle: {
		fontSize: theme.fontSize.large,
		fontWeight: '300',
		textAlign: 'center',
	},

	// summary
	summary: {
		paddingBottom: 40,
	},
	summaryText: {
		fontSize: theme.fontSize.default,
		fontWeight: '300',
		lineHeight: theme.fontSize.large,
		padding: theme.fontSize.large,
	},
});
