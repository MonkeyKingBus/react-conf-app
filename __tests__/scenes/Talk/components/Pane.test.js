import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Pane from '../../../../app/scenes/Talk/components/Pane';
import talks from '../../../../app/data/talks';

describe('Talk - Pane', () => {
	it('renders correctly', () => {
		const tree = renderer.create(
			<Pane
				nextTalk={talks[0]}
				onHeroLayout={() => {}}
				onScroll={() => {}}
				onScrollEndDrag={() => {}}
				prevTalk={talks[0]}
				showSpeakerModal={() => {}}
				visibleTalk={talks[0]}
			/>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
