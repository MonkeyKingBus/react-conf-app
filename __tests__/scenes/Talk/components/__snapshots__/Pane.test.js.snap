exports[`Talk - Pane renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "opacity": 0,
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 70,
              "justifyContent": "center",
              "left": 0,
              "paddingHorizontal": 60,
              "position": "absolute",
              "width": 750,
            },
            Object {
              "top": -70,
            },
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "textAlign": "center",
            }
          }>
          Keynote
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "textAlign": "center",
            }
          }>
          10:00am - Tom Occhino
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#383838",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomColor": "#CCC",
          "borderBottomWidth": 0.5,
          "borderTopColor": "#CCC",
          "borderTopWidth": 0.5,
          "marginTop": -0.5,
          "paddingBottom": 32,
          "paddingHorizontal": 24,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 32,
              "paddingTop": 32,
            }
          }>
          <View
            style={
              Object {
                "backgroundColor": "#F4F4F4",
                "borderRadius": 44,
                "height": 44,
                "overflow": "hidden",
                "width": 44,
              }
            }>
            <Image
              source={
                Object {
                  "uri": "https://www.gravatar.com/avatar/30840890815ed0044146eb2da203276e?s=128",
                }
              }
              style={
                Object {
                  "height": 44,
                  "width": 44,
                }
              } />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#00A8D8",
                "fontSize": 17,
                "fontWeight": "500",
                "marginTop": 14,
              }
            }>
            Tom Occhino
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999",
                "fontSize": 12,
                "paddingBottom": 24,
              }
            }>
            (tap for more)
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "300",
            "textAlign": "center",
          }
        }>
        Keynote
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingBottom": 40,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "300",
            "lineHeight": 24,
            "padding": 24,
          }
        }>
        Keynote Part 1
      </Text>
    </View>
    <View
      style={
        Object {
          "opacity": 0,
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 70,
              "justifyContent": "center",
              "left": 0,
              "paddingHorizontal": 60,
              "position": "absolute",
              "width": 750,
            },
            Object {
              "bottom": -70,
            },
          ]
        }>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#383838",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "textAlign": "center",
            }
          }>
          Keynote
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "textAlign": "center",
            }
          }>
          10:00am - Tom Occhino
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
